app/controllers/tweets_controller.rb

```diff
@@ -53,21 +49,28 @@ def setup_tweet
   def show
     @replies = Tweet.get_replies(params[:id])
     @user = User.find(@tweet.user_id)
+    Visitor.find_or_create_by(tweet_id: @tweet.id, user_id: current_user.id) if current_user.id != @tweet.user.id
```

Comment:
@amnindersingh12 
Can we move this to tweet model? and implement like `tweet.add_visitor(current_user)`

Reply:
done